USAGE.md

```diff
@@ -0,0 +1,194 @@
+            <integer>150</integer>
+            <key>ScaleX</key>
+            <integer>75</integer>
+            <key>ScaleY</key>
+            <integer>75</integer>
+        </dict>
+        <key>include</key>
+        <array>
+            <string>A</string>
+            <string>B</string>
+        </array>
+      </dict>
+    </array>
+```
+
+## Feature writing
+
+In a similar vein to the filter classes, `fontmake` allows you to customize the way that kerning and anchor rules in the font sources are turned into explicit rules in the autogenerated feature file. These generated rules are written by classes called feature writers. The feature writers can also be customized with a lib key, `com.github.googlei18n.ufo2ft.featureWriters`.
+
+For example, all feature writers take the `mode` option, which takes either the value `append` or `skip`. The default is `skip`, which will skip writing the feature if the feature is already explicitly present in the design sources' features file. To change this to `append` for the kern feature writer (i.e. to add generated kerning rules from the kerning table onto the end of the manually supplied `kern` feature), you would add the following lib key:
+
+```xml
+    <key>com.github.googlei18n.ufo2ft.featureWriters</key>
+    <array>
+        <dict>
+            <key>class</key>
+            <string>KernFeatureWriter</string>
+            <key>options</key>
+            <dict>
+                <key>mode</key>
+                <string>append</string>
+            </dict>
+        </dict>
+    </array>
+```
+
+`ufo2ft` provides three feature writer classes:
+
+* `GdefFeatureWriter` generates the `table GDEF { } GDEF;` statement, based on the categories of the glyphs (stored in the `public.openTypeCategories` lib key of the source font) and ligature caret anchors (anchors starting with `caret_` or `vcaret_`). It has no customizable parameters. It can be disabled with the `--no-generate-GDEF` flag.
+
+* `KernFeatureWriter` generates kerning features (or, for certain complex scripts, `dist` features) based on the kerning information in the design sources. It has two optional parameters in addition to `mode`: `ignoreMarks`, which defaults to `True`, will emit an `LookupFlag IgnoreMarks` in the generated `kern` feature; setting this to `False` will generate kern rules which do not ignore mark glyphs. Additionally, `quantization` can be set to an integer value to round the kern rules to the nearest multiple of its value, which can help with compressing the tables.
+
+* `MarkFeatureWriter` generates `mark` and `mkmk` features based on the anchor information in the design sources. It has one optional parameter, `quantization`, which rounds the anchor positions to the nearest multiple of its value, which makes anchors more likely to be shared in the `GPOS` table, potentially reducing its size at the expense of some fidelity.
+
+It is not possible to customize the feature writers from the command line.
```

Comment:
actually there is a `--feature-writers` CLI option, and works similarly to the --filters one.
It uses ufo2ft's `loadFeatureWritersFromString`:
https://github.com/googlefonts/ufo2ft/blob/7000301d24bc61a57d97fa5c2ed241ce6e25babe/Lib/ufo2ft/featureWriters/__init__.py#L96-L124